x-pack/plugins/security_solution/public/management/common/routing.ts

```diff
@@ -34,76 +35,49 @@ const querystringStringify: <ExpectedType extends object, ArgType>(
-      url += generatePath(MANAGEMENT_ROUTING_ENDPOINTS_PATH, {
-        pageName: SiemPageName.management,
-        tabName: ManagementSubTab.endpoints,
-      });
-    } else {
-      url += generatePath(MANAGEMENT_ROUTING_ROOT_PATH, {
-        pageName: SiemPageName.management,
-      });
-    }
+export const getEndpointListPath = (
+  props: { name: 'default' | 'endpointList' } & HostIndexUIQueryParams,
+  search?: string
+) => {
+  const { name, ...queryParams } = props;
+  const urlQueryParams = querystringStringify<HostIndexUIQueryParams, typeof queryParams>(
+    queryParams
+  );
+  const urlSearch = `${urlQueryParams && !isEmpty(search) ? '&' : ''}${search ?? ''}`;
 
-    if (urlQueryParams) {
-      url += `?${urlQueryParams}`;
-    }
-  } else if (props.name === 'endpointDetails' || props.name === 'endpointPolicyResponse') {
-    const { name, excludePrefix, ...queryParams } = props;
-    queryParams.show = (props.name === 'endpointPolicyResponse'
-      ? 'policy_response'
-      : '') as HostIndexUIQueryParams['show'];
-
-    url += `${generatePath(MANAGEMENT_ROUTING_ENDPOINTS_PATH, {
-      pageName: SiemPageName.management,
+  if (name === 'endpointList') {
+    return `${generatePath(MANAGEMENT_ROUTING_ENDPOINTS_PATH, {
       tabName: ManagementSubTab.endpoints,
-    })}?${querystringStringify<EndpointDetailsUrlProps, typeof queryParams>(queryParams)}`;
-  } else if (props.name === 'policyList') {
-    url += generatePath(MANAGEMENT_ROUTING_POLICIES_PATH, {
-      pageName: SiemPageName.management,
-      tabName: ManagementSubTab.policies,
-    });
-  } else if (props.name === 'policyDetails') {
-    url += generatePath(MANAGEMENT_ROUTING_POLICY_DETAILS_PATH, {
-      pageName: SiemPageName.management,
-      tabName: ManagementSubTab.policies,
-      policyId: props.policyId,
-    });
+    })}${appendSearch(`${urlQueryParams ? `${urlQueryParams}${urlSearch}` : urlSearch}`)}`;
   }
+  return `${appendSearch(`${urlQueryParams ? `${urlQueryParams}${urlSearch}` : urlSearch}`)}`;
+};
 
-  return url;
+export const getEndpointDetailsPath = (
+  props: { name: 'endpointDetails' | 'endpointPolicyResponse' } & EndpointDetailsUrlProps,
+  search?: string
+) => {
+  const { name, ...queryParams } = props;
+  queryParams.show = (props.name === 'endpointPolicyResponse'
+    ? 'policy_response'
+    : '') as HostIndexUIQueryParams['show'];
+  const urlQueryParams = querystringStringify<EndpointDetailsUrlProps, typeof queryParams>(
+    queryParams
+  );
+  const urlSearch = `${urlQueryParams && !isEmpty(search) ? '&' : ''}${search ?? ''}`;
+
+  return `${generatePath(MANAGEMENT_ROUTING_ENDPOINTS_PATH, {
+    tabName: ManagementSubTab.endpoints,
+  })}${appendSearch(`${urlQueryParams ? `${urlQueryParams}${urlSearch}` : urlSearch}`)}`;
 };
+
+export const getPoliciesPath = (search?: string) =>
+  `${generatePath(MANAGEMENT_ROUTING_POLICIES_PATH, {
+    tabName: ManagementSubTab.policies,
+  })}${appendSearch(search ?? undefined)}`;
```

Comment:
not sure `??` is needed here (or below in the next method). `search` will either be a `string` or `undefined`